lib/Refactoring/LocalRename.cpp

```diff
@@ -228,7 +230,17 @@ class RenameRangeCollector : public IndexDataConsumer {
   bool finishDependency(bool isClangModule) override { return true; }
 
   Action startSourceEntity(const IndexSymbol &symbol) override {
-    if (symbol.USR != usr) {
+    // We need to check both the USR and the display name because multiple
+    // distinct decls might have the same USR. For example, the following
+    // declarations have the same USR with a single ERROR_TYPE parameter despite
+    // being distinct declarations.
+    //
+    // func bar(body: Invalid) {}
+    // func bar(ignoreCase: Bool, body: Invalid) {}
+    //
+    // The display name contains the argument labels, which is sufficient to
+    // distinguish these declarations.
+    if (symbol.USR != usr || symbol.name != displayName) {
```

Comment:
Do we even need to compare USRs here in the first place? `IndexSymbol` contains a reference to the decl, and `ReferenceCollector` gets given the decl on construction (FWIW looks like the USR constructor is never used). Could we just compare decl pointers here?

Reply:
The original idea for using the USR was that this way local rename and global rename both operate on the same principle (comparing USRs). But maybe it’s time to give this up if we’ve discovered that USR comparison isn’t sufficient anyway.

Reply:
Changed it to check the `Decl` pointers for equality